assets/js/RenameColumns.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react'
+import {store, setSelectedWfModuleAction} from "./workflow-reducer";
+import {getPageID} from './utils'
+import {findModuleWithIdAndIdName, findParamValByIdName, getWfModuleIndexfromId} from "./utils";
+import WorkBenchAPI from './WorkbenchAPI'
+
+var api = WorkBenchAPI();
+export function mockAPI(mock_api) {
+    api = mock_api;
+}
+
+//renameInfo format: {prevName: <current column name in table>, newName: <new name>}
+
+function updateRenameModule(module, renameInfo, isNew=false) {
+    var entriesParam = findParamValByIdName(module, 'rename-entries');
+    var existingEntries = {}
+    try {
+        existingEntries = JSON.parse(entriesParam.value.trim());
+    } catch(e) {}
+    // If "prevName" in renameInfo exists as a value in edit entries,
+    // update that entry (since we are renaming a renamed column)
+    var entryExists = false;
+    for(let k in existingEntries) {
+        if(existingEntries[k] == renameInfo.prevName) {
+            existingEntries[k] = renameInfo.newName;
+            entryExists = true;
+            break;
+        }
+    }
+    // Otherwise, add the new entry to existing entries.
+    if(!entryExists) {
+        existingEntries[renameInfo.prevName] = renameInfo.newName;
+    }
+    if(isNew) {
+        var showAllParam = findParamValByIdName(module, 'display-all');
+        try {
+            api.onParamChanged(showAllParam.id, {value: false})
+                .then(api.onParamChanged(entriesParam.id, {value: JSON.stringify(existingEntries)}));
+        } catch(e) {}
+    }
+    else {
+        api.onParamChanged(entriesParam.id, {value: JSON.stringify(existingEntries)});
+    }
+}
+
+export function updateRename(wfModuleId, renameInfo) {
```

Comment:
This should clearly be a reducer. But I see that we use this exact pattern in all our other hard-coded DataGrid-related modules, so I guess we can defer on fixing that.